### DISCUSSION TITLE
Why can't the setup be renamed?

### DISCUSSION THREAD

**Discussion by NurudinImsirovic** [sumatrapdfreader/sumatrapdf#2793] [General]
Renaming **SumatraPDF-3.4.6-64-install.exe** to **setup.exe** causes an error:

```
[Window Title]
SumatraPDF installer

[Main Instruction]
Looks like corrupted installation of SumatraPDF.


[Content]
Learn more at www.sumatrapdfreader.org/docs/Corrupted-installation.

[Close]
```

My naming convention for setup files looks like this:

**DRIVE-LETTER**/download/archive/software/**CATEGORY**/**NAME**/**setup.exe**

My directory structure for all setups looks like this:

```
drive-d/download/archive/software/office/sumatrapdf/SumatraPDF-3.4.6-64-install.exe
drive-d/download/archive/software/text-editor/visual-studio-code/setup.exe
drive-d/download/archive/software/web-browser/google-chrome/setup.exe
...
```

As you can see having that one outlier isn't very consistent alongside with all the other files.. maybe I suffer from OCD or something.

I could use the Portable version, but then I have to manually assign Default application (or maybe not?) for each file type SumatraPDF can read.

**Comment by GitHubRulesOK** [comment]
using an installer flag in this way is not uncommon see the portable MikTex "download the standard installer and rename it to miktex-portable.exe" or more similarly the dual console/gui exif tool `exiftool(-k).exe` 

Many people misused the files so during install the command line must include that flag `setup -install -blah -blah` see `-help`. The easiest means for "one exe" distribution is include the flag in the filename. You can remove it but during install you need to include the required `-install` switch 
![image](https://user-images.githubusercontent.com/12754913/178103130-80d5a1c4-57c6-4f05-96af-e067c79d8a00.png)
![image](https://user-images.githubusercontent.com/12754913/178103217-ea0389dc-a969-4359-be17-2a3abf63334e.png)

**Comment by kjk** [comment]
Installer and Sumatra executable is the same .exe. It acts as installer if there's `installer` in the file name or if you use `-install` cmd-line flags.

**Comment by NurudinImsirovic** [comment]
Thank you @GitHubRulesOK and @kjk. Everything's set up now 👍
